contracts/rewards/src/msg.rs

```diff
@@ -0,0 +1,58 @@
+use axelar_wasm_std::{nonempty, Threshold};
+use cosmwasm_schema::cw_serde;
+
+#[cw_serde]
+pub struct InstantiateMsg {
+    governance_address: String,
+    rewards_denom: String,
+    params: RewardsParams,
+}
+
+#[cw_serde]
+pub struct RewardsParams {
+    /// How often rewards are calculated, specified in number of blocks. Participation is calculated over this window. So if epoch is 500
+    /// blocks, workers are rewarded for their participation within each 500 block window.
+    epoch_duration: nonempty::Uint64,
+
+    /// Total length of time over which rewards in a pool are distributed, specified in number of blocks. For example, if pool_duration is
+    /// 1 million blocks, and epoch_duration is 500 blocks, and total size of the reward pool is 100k tokens, every 500 blocks,
```

Comment:
Does "total size of the reward pool" change every window as rewards are given out and are added? How does the calculation work when more rewards are added while the pool isn't empty.

Reply:
Calculation would have to change as more rewards are added. Pretty sure the calculation then is:
```
total rewards remaining / epochs remaining / participating validators
```
where total rewards remaining is rewards not yet distributed and epochs remaining is number of epochs for which rewards have not yet been distributed. I can update the comment to this if you like, but I do feel like it's a little harder to wrap your head around this than the basic case example.

Reply:
What does epochs remaining mean? I don't see any param related to epochs when adding rewards.

Reply:
In the params, there is an `epoch_duration`

Reply:
The current epoch is just determined from the current block height, so we don't really need it for any of the messages.